### PR TITLE
types (GLTFLoader): add definitions about plugin system

### PR DESCRIPTION
**Pull request by 0b5vr**
Add type definitions about GLTFLoader plugin system which is introduced in #19144 .

### Points need review

- I used a name `GLTFLoaderPlugin` as an interface name, which is not explicitly mentioned in anywhere yet. is the name appropriate?
- I'm not sure I should have `_markDefs` in this definition
- I don't have enough confidence about return type of:
    - `getMaterialType`, returns `typeof Material | null`.
    - `extendMaterialParams`, returns `Promise<any>`
    - These are working fine in my WIP project at least

### CODE REVIEW

File: examples/jsm/loaders/GLTFLoader.d.ts [hunk 1]
```diff
@@ -39,7 +41,11 @@ export class GLTFLoader extends Loader {
 	load( url: string, onLoad: ( gltf: GLTF ) => void, onProgress?: ( event: ProgressEvent ) => void, onError?: ( event: ErrorEvent ) => void ) : void;
 	setDRACOLoader( dracoLoader: DRACOLoader ): GLTFLoader;
 	setDDSLoader( ddsLoader: DDSLoader ): GLTFLoader;
-	setKTX2Loader( ktx2Loader: KTX2Loader ): GLTFLoader;
+
+	register( callback: ( parser: GLTFParser ) => GLTFLoaderPlugin ): GLTFLoader;
+	unregister( callback: ( parser: GLTFParser ) => GLTFLoaderPlugin ): GLTFLoader;
+
+  setKTX2Loader( ktx2Loader: KTX2Loader ): GLTFLoader;
```
Comment (mrdoob) [inline examples/jsm/loaders/GLTFLoader.d.ts:48]: Nit: Indent

Comment (FMS-Cat) [reply examples/jsm/loaders/GLTFLoader.d.ts:48]: thanks GitHub editor very cool

Comment (FMS-Cat) [reply examples/jsm/loaders/GLTFLoader.d.ts]: fixed @ e5cb0c9

**Review by donmccurdy — APPROVED** [review]
looks good, thank you!

**Comment by donmccurdy** [comment]
> I'm not sure I should have _markDefs in this definition...

It might be better to hide this one.. i'd be OK with it being a public method, if someone eventually needed it, but if so it should not be prefixed with an underscore.

**Comment by takahirox** [comment]
Yes, `_markDefs` should be hidden (or we should remove the prefix '_').

When I implemented then lights extension with the plugin system, I couldn't avoid to extend `_markDefs`. But the method is for a bit too internal use. So I'd like to come up with a nicer way and somehow remove `_markDefs` extension at some point...

**Comment by donmccurdy** [comment]
Let's omit it from type defs for now, then. We can un-prefix it and add it later, if it turns out we can't remove it and other extensions need it.

**Review by takahirox — APPROVED** [review]
Looking good to me (other than exposing _markDefs as commented)

**Comment by FMS-Cat** [comment]
thanks for the ping!

gonna delete the definition of `_markDefs`.

**Comment by FMS-Cat** [comment]
@mrdoob ready!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**